### PR TITLE
Check if event is cancelable before preventDefault

### PR DESCRIPTION
**Pull request by SychO9**
**Fixes flarum/core#2392**

### CODE REVIEW

**Review by askvortsov1 — APPROVED** [review]
Makes sense, thanks!

**Comment by clarkwinkelmann** [comment]
While we're at it, could we add a comment that explains what that code does?

I don't understand why there's a `preventDefault` there. With the change, in which situations is it called and in which situations is it not called?

**Comment by SychO9** [comment]
Looking at the code again and the behavior while testing, I'm actually pretty sure the call to `preventDefault()` on the `touchstart` event here does nothing.

To try and understand the wanted behavior behind the code, I've found these three commits:
**June 24, 2015**: 9e8050530e63a661a2d74a9ba7a0b9d1af1fa056
**July 28, 2015**: 2d09f661d027a0b3936bdb8367e4b8fbc50c721a
**July 28, 2015**: 29093d8aa4045313cc7612012338f48fc8739790

The **1st** commit suggests that the wanted behavior was to open the post mention dropdown preview when clicking it (on touchscreens) instead of directly going to the mentioned post. However that code resulted in also preventing the click behavior on dropdown items.


The **2nd** commit tries to fix the problem with clicking on the dropdown item from the 1st commit, but doesn't seem to actually fix it. (from tests)

The **3rd** commit recovers clicking behavior on the dropdown item, but loses the behavior of showing the dropdown when clicking the post mention instead of going to the actual post. (all on touchscreens of course)

Calling `preventDefault()` on a touchend event instead fixes the initial wanted behavior from the first commit.

File: js/src/forum/addPostMentionPreviews.js [hunk 1]
```diff
@@ -103,7 +103,15 @@ export default function addPostMentionPreviews() {
...
 
-      $this.on('touchstart', e => e.preventDefault());
+      // On touchscreens we don't want clicking on the post mention
+      // to take us to the actual post
+      // instead we want it to open the preview dropdown.
+      // And only by clicking on the preview dropdown item
+      // will it take us to the actual post.
+      $this.on('touchend', e => {
+        if (e.cancelable)
+          e.preventDefault();
```
Comment (askvortsov1) [inline js/src/forum/addPostMentionPreviews.js:113]: Could we wrap this in curly braces?

File: js/src/forum/addPostMentionPreviews.js [hunk 2]
```diff
@@ -103,7 +103,15 @@ export default function addPostMentionPreviews() {
         }
       };
 
-      $this.on('touchstart', e => e.preventDefault());
+      // On touchscreens we don't want clicking on the post mention
+      // to take us to the actual post
+      // instead we want it to open the preview dropdown.
+      // And only by clicking on the preview dropdown item
+      // will it take us to the actual post.
```
Comment (askvortsov1) [inline js/src/forum/addPostMentionPreviews.js:110]:
```suggestion
      // If on a touch (mobile) device, open the post preview on click.
      // Users can then click on the preview to go to the post if desired.
```

**Review by clarkwinkelmann — COMMENTED** [review]
Looking good! But I still think it would make sense to mention in the comment that the intent of the code is to prevent navigation to the link. Maybe something like:

```
      // On a touch (mobile) device we cannot hover the link to reveal the preview.
      // Instead we cancel the navigation so that a click reveals the preview.
      // Users can then click on the preview to go to the post if desired. 
```

I see you have already revised that comment multiple times, I hope this suggestion is not a duplicate of the previous discussion :innocent:

**Comment by SychO9** [comment]
that looks good to me!

**Review by clarkwinkelmann — APPROVED** [review]

**Pull request merged**